### PR TITLE
Function headers rpc

### PR DESCRIPTION
**Pull request by Sniperhid**
**When merged this pull request will:**
- Document the headers for sys_sounds and sys_rpc. Complete with examples and text descriptions.
- Some small progress towards #107

### CODE REVIEW

File: addons/sys_rpc/fnc_addProcedure.sqf [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Procedure name <STRING>
+ * 1: Callback code <CODE>
  *
  * Return Value:
- * RETURN VALUE <TYPE>
+ * NONE
```
Comment (jonpas) [inline addons/sys_rpc/fnc_addProcedure.sqf]: `None`

File: addons/sys_rpc/fnc_handleData.sqf [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
+ * Handles the receipt of messages from the teamspeak plugin. Firstly by de-serializing the recieved message. It will then call any procedure handlers.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Message <STRING>
  *
  * Return Value:
- * RETURN VALUE <TYPE>
+ * Always returns true <BOOL>
```
Comment (jonpas) [inline addons/sys_rpc/fnc_handleData.sqf]: This is not true, L22 has no return value. Should probably change that to return `false` and have return value as `* Successfully handled <BOOL>`

File: addons/sys_sounds/fnc_handleLoadedSound.sqf [hunk 3]
```diff
@@ -1,16 +1,16 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * This is the callback function for recieving the event "handleLoadedSound" from teamspeak. It just ensures a sound is marked as loaded to prevent it from being loaded again. It will also call the callback function specified in the loadSound function if one exists.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio classname <STRING>
+ * 1: Load code <NUMBER> - 1 represents a successful load.
```
Comment (jonpas) [inline addons/sys_sounds/fnc_handleLoadedSound.sqf]: `* 1: Load code (1 represents a successful load) <NUMBER>`

File: addons/sys_rpc/fnc_handleData.sqf [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
  *
  * Example:
- * [ARGUMENTS] call acre_COMPONENT_fnc_FUNCTIONNAME
+ * ["localStopSpeaking:1,0,'',"] call acre_sys_rpc_fnc_handleData
```
Comment (jonpas) [inline addons/sys_rpc/fnc_handleData.sqf:12]: Adding space between each argument will improve readability here.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_rpc/fnc_handleData.sqf (hunk 4 above)
Comment (Sniperhid) [reply addons/sys_rpc/fnc_handleData.sqf:12]: Note this is a single string as it would be recieved from the plugin...

Comment (jonpas) [reply addons/sys_rpc/fnc_handleData.sqf:12]: Oh, disregard then, I am blind.

File: addons/sys_rpc/fnc_handleData.sqf [hunk 5]
```diff
@@ -6,7 +6,7 @@
  * 0: Message <STRING>
  *
  * Return Value:
- * Always returns true <BOOL>
+ * Handled - Returns true if a handler was called. <BOOL>
```
Comment (jonpas) [inline addons/sys_rpc/fnc_handleData.sqf]: `* Handled (returns true if a handler was called) <BOOL>`

Full stops in lists look ugly.

**Review by jonpas — APPROVED** [review]

**Pull request merged**